flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/table/restrictions/BigQueryPartitionTest.java

```diff
@@ -179,166 +181,311 @@ public void testWrongNumericPartition() {
-        Assertions.assertThat(
-                        BigQueryPartition.formatPartitionRestrictionBasedOnInfo(
-                                Optional.of(info), "event", "2010-10-10 10:10:10"))
-                .isEqualTo("event BETWEEN '2010' AND '2011'");
+    public void testCheckPartitionCompletedHour() {
+        PartitionIdWithInfo partitionWithInfo =
+                new PartitionIdWithInfo(
+                        "2023072801",
+                        new TablePartitionInfo(
+                                "temporal",
+                                BigQueryPartition.PartitionType.HOUR,
+                                StandardSQLTypeName.TIMESTAMP,
+                                // partition watermark is now
+                                Instant.now()));
+
+        PartitionIdWithInfoAndStatus partitionWithInfoAndStatus =
+                BigQueryPartition.checkPartitionCompleted(partitionWithInfo);
+
+        Assertions.assertThat(partitionWithInfoAndStatus.getStatus())
+                .isEqualTo(BigQueryPartition.PartitionStatus.COMPLETED);
     }
 
     @Test
-    public void testDateFormatDayPartitionRestriction() {
-        TablePartitionInfo info =
-                new TablePartitionInfo(
-                        "event", BigQueryPartition.PartitionType.DAY, StandardSQLTypeName.DATE);
-        Assertions.assertThat(
-                        BigQueryPartition.formatPartitionRestrictionBasedOnInfo(
-                                Optional.of(info), "event", "2010-10-10"))
-                .isEqualTo("event BETWEEN '2010-10-10' AND '2010-10-11'");
+    public void testCheckPartitionCompletedDay() {
+        PartitionIdWithInfo partitionWithInfo =
+                new PartitionIdWithInfo(
+                        "20230728",
+                        new TablePartitionInfo(
+                                "temporal",
+                                BigQueryPartition.PartitionType.DAY,
+                                StandardSQLTypeName.DATE,
+                                // partition watermark is now
+                                Instant.now()));
+
+        PartitionIdWithInfoAndStatus partitionWithInfoAndStatus =
+                BigQueryPartition.checkPartitionCompleted(partitionWithInfo);
+
+        Assertions.assertThat(partitionWithInfoAndStatus.getStatus())
+                .isEqualTo(BigQueryPartition.PartitionStatus.COMPLETED);
     }
 
     @Test
-    public void testDateFormatMonthPartitionRestriction() {
-        TablePartitionInfo info =
-                new TablePartitionInfo(
-                        "event", BigQueryPartition.PartitionType.MONTH, StandardSQLTypeName.DATE);
-        Assertions.assertThat(
-                        BigQueryPartition.formatPartitionRestrictionBasedOnInfo(
-                                Optional.of(info), "event", "2010-10-10"))
-                .isEqualTo("event BETWEEN '2010-10' AND '2010-11'");
+    public void testCheckPartitionCompletedMonth() {
```

Comment:
We should add edge case tests for February as leap month, non-leap month, a 30 day month, a 31 day month.

Reply:
Done

Reply:
Please add not completed 31 day month case. LGTM otherwise.

Reply:
Done